script.js

```diff
@@ -1,4 +1,447 @@
+    L2GameMode = L2_DECIDE_WINNER;
+    arrayForNums.push(playerTwoNum);
+  } else if ((currentPlayer == 2) & (arrayRolls[0] < arrayRolls[1])) {
+    playerTwoNum = selectBA();
+    myOutputValue = `Player ${currentPlayer}, your two dice rolls are: ${arrayRolls[0]} and ${arrayRolls[1]}. <br><br>Computer picked ${playerTwoNum} for you.`;
+    L2GameMode = L2_DECIDE_WINNER;
+    arrayForNums.push(playerTwoNum);
+  }
+  arrayRolls = [];
+  return myOutputValue;
+};
+
+//_____________________________ Variable number of Dice Mode_________________
+
+// will only call out the differences between this mode and Base Normal mode.
+
+var variableDice = function (input) {
+  var myOutputValue = "";
+  if (L2GameMode == L2_CHOOSE_NUM_DICE) {
+    myOutputValue = diceNum(input);
+  } else if (L2GameMode == L2_ROLL_DICE) {
+    myOutputValue = rollMultipleDice(input);
+  } else if (L2GameMode == L2_DECIDE_WINNER) {
+    myOutputValue = decideWinner(input);
+  }
+  return myOutputValue;
+};
+
+// create new function for user to select number of dice.
+
+var diceNum = function (input) {
+  numofDice = input;
+  var myOutputValue = `You have choosen to play with ${input} dice 🎲. Hit Submit to start rolling!`;
+  console.log("numofDice", numofDice);
+  L2GameMode = L2_ROLL_DICE;
+  return myOutputValue;
+};
+
+// adapted from rollAndDecide function, but not it catered for multiple dice scenarios.
+// used sort array to get the best order.
+
+var rollMultipleDice = function () {
+  var myOutputValue = "";
+  var counterA = 0;
+  var line1 = "";
+  var line2 = "";
+  while (counterA < numofDice) {
+    arrayRolls.push(diceRoll());
+    counterA += 1;
+  }
+  line1 = `Player ${currentPlayer}, your dice rolls are: ${arrayRolls}.<br><br>`;
+  arrayRolls = arrayRolls.sort((a, b) => b - a);
+  var bestNumber = "";
+  var counterB = 0;
+  while (counterB < arrayRolls.length) {
+    bestNumber = bestNumber + `${arrayRolls[counterB]}`;
+    counterB += 1;
+  }
+  if (currentPlayer == 1) {
+    playerOneNum = Number(bestNumber);
+    line2 = `Computer picked ${playerOneNum} for you.`;
+    L2GameMode = L2_ROLL_DICE;
+    currentPlayer = 2;
+    arrayForNums.push(playerOneNum);
+  } else if (currentPlayer == 2) {
+    playerTwoNum = Number(bestNumber);
+    line2 = `Computer picked ${playerTwoNum} for you.`;
+    L2GameMode = L2_DECIDE_WINNER;
+    arrayForNums.push(playerTwoNum);
+  }
+  myOutputValue = line1 + line2;
+  arrayRolls = [];
+  return myOutputValue;
+};
+
+// _________________ Variable number of players ____________________
+
+// this part is not fully completed yet
+
+var variablePlayers = function (input) {
+  var myOutputValue = "";
+  if (L2GameMode == L2_CHOOSE_NUM_DICE) {
+    myOutputValue = diceNum(input);
+  } else if (L2GameMode == L2_ROLL_DICE) {
+    myOutputValue = rollMultipleDice(input);
+  } else if (L2GameMode == L2_DECIDE_WINNER) {
+    myOutputValue = decideWinner(input);
+  }
+  return myOutputValue;
+};
```

Comment:
It's great decomposition again; isolating different game modes with the relevant messages